darkmoon/api/v1/metadata/views.py

```diff
@@ -280,18 +312,32 @@ async def upload_metadata(
                     "operating_system": data_type[4],
                 },
             }
-            await collection.update_one(duplicate_hashes, change)
-            return UploadMetadataResponse(
-                message="Successfully Updated Object.",
-                data=file,
-            )
+            if any("header_info" in item for item in file):
+                await collection.update_one(duplicate_hashes, change)
+                return UploadMetadataResponse(
+                    message="Successfully Updated Object.",
+                    data=EXEMetadata.parse_obj(file_metadata),
+                )
+            else:
+                await collection.update_one(duplicate_hashes, change)
```

Comment:
The cognitive complexity of this area of code is really high. We should consider refactoring it into a simpler algorithm or into functions.